test/client/getOrderbook.ts

```diff
@@ -72,142 +55,136 @@ function normalRippledResponse(
-//     return rippled.book_offers.xrp_usd
-//   }
-//   if (request.taker_gets.issuer === 'rp8rJYTpodf8qbSCHVTNacf8nSW8mRakFw') {
-//     return rippled.book_offers.usd_xrp
-//   }
-//   throw new Error('unexpected end')
-// }
+function xrpRippledResponse(
+  request: BookOffersRequest,
+): Record<string, unknown> {
+  if (request.taker_pays.issuer === 'rp8rJYTpodf8qbSCHVTNacf8nSW8mRakFw') {
+    return rippled.book_offers.xrp_usd
+  }
+  if (request.taker_gets.issuer === 'rp8rJYTpodf8qbSCHVTNacf8nSW8mRakFw') {
+    return rippled.book_offers.usd_xrp
+  }
+  throw new Error('unexpected end')
+}
 
 describe('client.getOrderbook', function () {
   beforeEach(setupClient)
   afterEach(teardownClient)
 
-  addressTests.forEach(function (testcase) {
-    describe(testcase.type, function () {
-      it('normal', async function () {
-        this.mockRippled.addResponse('book_offers', normalRippledResponse)
-        const response = await this.client.getOrderbook(
-          requests.getOrderbook.normal.taker_pays,
-          requests.getOrderbook.normal.taker_gets,
-          { limit: 1 },
-        )
-        assert.deepEqual(response, responses.getOrderbook.normal)
-      })
+  it('normal', async function () {
+    this.mockRippled.addResponse('book_offers', normalRippledResponse)
+    const response = await this.client.getOrderbook(
+      requests.getOrderbook.normal.takerPays,
+      requests.getOrderbook.normal.takerGets,
+      {
+        limit: 1,
+      },
+    )
+    const expectedResponse = {
+      buy: responses.getOrderbook.normal.buy.slice(0, 1),
+      sell: responses.getOrderbook.normal.sell.slice(0, 1),
+    }
+    assertResultMatch(response, expectedResponse, 'getOrderbook')
+  })
+
+  it('invalid options', async function () {
+    this.mockRippled.addResponse('book_offers', normalRippledResponse)
+    assertRejects(
+      this.client.getOrderbook(
+        requests.getOrderbook.normal.takerPays,
+        requests.getOrderbook.normal.takerGets,
+        {
+          invalid: 'options',
+        },
+      ),
+      this.client.errors.ValidationError,
+    )
+  })
+
+  it('with XRP', async function () {
+    this.mockRippled.addResponse('book_offers', xrpRippledResponse)
+    const response = await this.client.getOrderbook(
+      requests.getOrderbook.withXRP.takerPays,
+      requests.getOrderbook.withXRP.takerGets,
+    )
+    assertResultMatch(response, responses.getOrderbook.withXRP, 'getOrderbook')
+  })
+
+  it('sample USD/XRP book has orders sorted correctly', async function () {
+    this.mockRippled.addResponse('book_offers', xrpRippledResponse)
+    const response = await this.client.getOrderbook(
+      requests.getOrderbook.withXRP.takerPays,
+      requests.getOrderbook.withXRP.takerGets,
+    )
+    checkSortingOfOrders(response.buy)
+    checkSortingOfOrders(response.sell)
+  })
+
+  it('sorted so that best deals come first [bad test]', async function () {
```

Comment:
nit: "bad test" might be interpreted as 'this test has poor coverage' instead of 'this test is expected to throw an error'. Maybe better wording would be `failure case` (Whatever you decide, apply it to the other test names :) )